src/Plugin/jsonapi/FieldEnhancer/WebformConfirmationEnhancer.php

```diff
@@ -30,7 +31,17 @@ protected function doUndoTransform($data, Context $context) {
         ->loadByProperties(['uuid' => $data]);
       if (!empty($webform_submission_by_uuid)) {
         $webform_submission_by_uuid = reset($webform_submission_by_uuid);
-        return $webform_submission_by_uuid->getElementData('confirmation_message');
+        $message = $webform_submission_by_uuid->getElementData('confirmation_message');
+        // We only check if the [webform:handler token exists or not, as it may
+        // involve communicating outside if it is still in here, meaning the
+        // communication gets failed.
+        $is_unprocessed_token = (strpos(print_r($message, TRUE), '[webform:handler:') !== FALSE) ? TRUE : FALSE;
+        if ($is_unprocessed_token) {
+          // Removes the failed webform_submission.
+          $webform_submission_by_uuid->delete();
+          throw new NotFoundHttpException();
```

Comment:
Is this the right exception to throw here? I guess it's 404?

Reply:
yeah, just a 404 response. FE doesn't care about what it gets. just return a `please try again` message to users.
![image](https://user-images.githubusercontent.com/8788145/204454557-f385ebc5-98ec-497b-a182-80d19a7420ee.png)

Reply:
Thanks for the CR ! @anthony-malkoun

Reply:
As discussed, this probably needs a separate ticket (and as @vincent-gao mentioned this is almost just a quick PoC) to capture the right behaviour. Does the frontend (and maybe optionally?) need to see the actual message from the POST or should it just be a generic message? In general though this approach is fine.

Reply:
hey @anthony-malkoun 
now, the BE will forward the message of remote response following `{code}|{message}` format. For more details see the updated description of this PR.
I think it needs a FE ticket for handling the message. @MdNadimHossain